clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -284,6 +284,9 @@ def AllocaOp : CIR_Op<"alloca", [
     cases, the first use contains the initialization (a cir.store, a cir.call
     to a ctor, etc).
 
+    The `dynAllocSize` is a number of elements to be allocated on the stack.
+    Once not specified, the number of elements allocated is one.
```

Comment:
When the allocation is done without a dynamic size, we compute the actual alloca based on the ABI size for the type. It seems to me that when `dynAllocSize` is specified, clang don't really care about the type at hand. Assuming this is right, this should probably be written as:

```
The `dynAllocSize` specifies the size to dynamically allocate on the stack and ignores the allocation size based on the original type. This is useful when handling VLAs and is omitted when declaring regular local variables.
```